contracts/rosca.sol

```diff
@@ -1,266 +1,258 @@
+    membersAddresses.push(msg.sender);
+    numRoundsInEpoch = 1;
+  }
+
+  /** startRound() check if the round has ended, if so, get the winner of the currentRound's pot
+    * and add the amount to the winner's pending withdrawal
+    * if there were no bid during the round, address at index "currentRoundNumber" is chosen,
+    * if the address chosen already won the pot once in currentEpoch, currentRoundNumber+1 is chosen, untill a winner is found
+    *
+    * after the winner had been declared, bidding process reset and currentRound is incremented, if currentRound+1 is equal numRoundsInEpoch,
+    * currentEpoch is incremented, and currentRound = 1;
+    *
+    * if currentRound = 0(ROSCA hasnt started), no winner is declared
+    */
+  function startRound()
+  {
+
+    if(now < startTime + (uint(currentRound + (currentEpoch * numRoundsInEpoch))  * (uint(roundPeriodInDays) * 1 days)))
+      throw;
+    if(currentRound != 0 ){
+      if(winnerAddress == 0)
+      {
+        //there is no bid in this round so find an unpaid address for this epoch
+        for(uint i = 0 ; i < membersAddresses.length; i++)
         {
-            if(currentRound != 0 ){
-                if(winnerAddress == 0)
-                {
-                    //there is no bid in this round so find an unpaid address for this epoch
-                    for(uint i = 0 ; i < 10; i++)
-                    {
-                        if(members[memberAddress[(currentRound-1 + i) % memberAddress.length]].paid < currentEpoch)
-                        winnerAddress = memberAddress[currentRound-1 + i];
-                    }
-
-                }
-                members[winnerAddress].allowance += lowestBid;
-                members[winnerAddress].paid++;
-                LogRoundFundsRelease(winnerAddress, lowestBid);
-
-            }
-            if(currentRound < numRounds)  // reset variables related to bidding
-            {
-                lowestBid = roundSum;
-                winnerAddress = 0;
-                // TODO: set winnderAddress to a random member that havent got paid in this epoch
-
-                currentRound++;
-            }else if(currentEpoch < numEpochs) //end of Epoch reset rounds
-            {
-                currentEpoch++;
-                currentRound = 1;
-                lowestBid = roundSum;
-
-            }
-
-
+          if(members[membersAddresses[(currentRound-1 + i) % membersAddresses.length]].paid < currentEpoch)
+          winnerAddress = membersAddresses[currentRound-1 + i];
         }
-
-    }
-    /**
-     * Registers another user-proxy for this ROSCA (msg.sender is the proxy).
-     */
-    function joinRequest() beforeStart
-    {
-        //only put the request in the pending list if they are not in the ROSCA already
-        if(!members[msg.sender].alive)
-            pendingJoinRequest[msg.sender] = true;
-        else
-            throw;
+      }
+      members[winnerAddress].pendingWithdrawl += lowestBid;
```

Comment:
The whole logic seems right to me, but as it's pretty complex with many if's and the stakes are high (this is real money), I'm a bit concerned abut this line - what if someone somehow manages to get to it twice? then they'll be "stealing" the win twice.
Can we, just for sake of security, double check and throw if needed, members[winnerAddress].paid before giving them money? 
I think it may be worth the extra gas.